google/cloud/aiplatform/pipeline_jobs.py

```diff
@@ -0,0 +1,316 @@
+                The key needs to be in the same region as where the compute
+                resource is created.
+
+                If this is set, then all
+                resources created by the BatchPredictionJob will
+                be encrypted with the provided encryption key.
+
+                Overrides encryption_spec_key_name set in aiplatform.init.
+            labels (Dict[str,str]):
+                Optional. The user defined metadata to organize PipelineJob.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to create this batch prediction
+                job. Overrides credentials set in aiplatform.init.
+            project (str),
+                Optional. Project to retrieve PipelineJob from. If not set,
+                project set in aiplatform.init will be used.
+            location (str),
+                Optional. Location to create PipelineJob. If not set,
+                location set in aiplatform.init will be used.
+
+        Raises:
+            ValueError: If inputs are formatted wrong.
+        """
+        utils.validate_display_name(display_name)
+
+        super().__init__(project=project, location=location, credentials=credentials)
+
+        self._parent = initializer.global_config.common_location_path(
+            project=project, location=location
+        )
+        pipeline_root = pipeline_root or initializer.global_config.staging_bucket
+        pipeline_spec = json_utils.load_json(
+            template_path, self.project, self.credentials
+        )
+        pipeline_name = pipeline_spec["pipelineSpec"]["pipelineInfo"]["name"]
+        job_id = job_id or "{pipeline_name}-{timestamp}".format(
+            pipeline_name=re.sub("[^-0-9a-z]+", "-", pipeline_name.lower())
+            .lstrip("-")
+            .rstrip("-"),
+            timestamp=_get_current_time().strftime("%Y%m%d%H%M%S"),
+        )
+        if not _VALID_NAME_PATTERN.match(job_id):
+            raise ValueError(
+                "Generated job ID: {} is illegal as a Vertex pipelines job ID. "
+                "Expecting an ID following the regex pattern "
+                '"[a-z][-a-z0-9]{{0,127}}"'.format(job_id)
+            )
+
+        job_name = _JOB_NAME_PATTERN.format(parent=self._parent, job_id=job_id)
+
+        pipeline_spec["name"] = job_name
+        pipeline_spec["displayName"] = job_id
+
+        builder = pipeline_utils.PipelineRuntimeConfigBuilder.from_job_spec_json(
+            pipeline_spec
+        )
+        builder.update_pipeline_root(pipeline_root)
+        builder.update_runtime_parameters(parameter_values)
+
+        runtime_config = builder.build()
+        pipeline_spec["runtimeConfig"] = runtime_config
+
+        _set_enable_caching_value(pipeline_spec["pipelineSpec"], enable_caching)
+
+        if encryption_spec_key_name is not None:
+            pipeline_spec["encryptionSpec"] = {"kmsKeyName": encryption_spec_key_name}
+
+        if labels:
```

Comment:
This validation should be moved to the top so it fails before we possibly retrieve the template_path from GCS.

Reply:
Added. There's also a `job_id` that's left underneath, but I'm thinking of not moving that, since `job_id` is constructed later?